docs/v3/stores/filesystem/v1.0.rst

```diff
@@ -182,6 +182,19 @@ in the section above.
   directory path ``dp``.
 
 
+Canonical URI
+=============
+
+The canonical URI format for this store follows the file URI scheme defined in
+[RFC8089]_. In the common case where the hostname is omitted this is
+``"file:///" + P``, where `P` is the base directory path, e.g.
+``"file:///C:\\data\\foo\\bar"``.
```

Comment:
This should be `"file://" + P`, e.g. `file://c:\data\foo\bar` or `file:///home/user/path/to/data.zarr`

Also, the backslashes inside the double quoted url is a bit confusing here, because we haven't specified what escaping mechanism we are assuming for that double quoted string.

If we intend to later support more elaborate URL syntax, e.g. `file:///path/to/group.zarr#path/to/array`, then we also need to define how special characters like "#" are escaped.  Presumably we would want to use percent encoding, as is typical for URLs.  But means that the following comment:

> In cases where a filesystem store may be considered the default, the ``"file:///"`` prefix can be omitted and only the base directory path is used

is problematic, because for a bare path you would presumably not use percent encoding.

Reply:
>  e.g. file://c:\data\foo\bar or file:///home/user/path/to/data.zarr

It seems like for windows it should even be `file:///c:/data/foo/bar`.

> Also, the backslashes inside the double quoted url is a bit confusing here, because we haven't specified what escaping mechanism we are assuming for that double quoted string.

I was assuming that those need to be escaped for RST, but this is not the case for code formatted parts.

> If we intend to later support more elaborate URL syntax, e.g. file:///path/to/group.zarr#path/to/array, then we also need to define how special characters like "#" are escaped. Presumably we would want to use percent encoding.

Yes, this is even specified by the file URI RFC, as it relies on the default URI RFC.

I'll adapt those points and have a rather flexible definition of the form without the prefix.

Reply:
Ah okay, I didn't realize that an extra initial slash was used even on Windows.